External/SPMmodified/xASL/xASL_im_FillNaNs.m

```diff
@@ -110,39 +111,45 @@
     IM(IM==0) = NaN;
 end
 
+% Also catch Infs or imaginary numbers or whatever. Zeroes are already catched above in the main function.
+IM(~isfinite(IM)) = NaN;
```

Comment:
I considered this as well, but you don't want this. When we use this function to just remove NaNs in images, sometimes you may want imaginary numbers or Inf's to remain. What happens with Inf's or imaginary numbers that are not at the edges?

Reply:
I can remove this line if you want. But this is not my idea, this line is from **your code**, I just moved it here from another location, because in my version, I need to do this before running the subfunction.

Reply:
I know. I mean that now, this is run for all METHODS. Whereas we only want it for METHOD 3. Otherwise I'm fine.

Reply:
OK. I'll add it for method 3 only...

Reply:
Then just move it 1 line before checking for the presence of NaNs, either inside METHOD 3 or just outside it.

It's probably not very important, but we are not testing the use cases of methods 1 and 2 so they should not change, and now all methods get this change because you put it in the general image loading section.

Reply:
That's the problem. Before checking if we have method 1/2/3, we need to check if we have NaNs or not at all. So the INF->NAN has to happen before that. Or else we first check there are no NaNs, so decide to skip the filling, but leave INFs. 

Please check the new code how it looks after all changes -> it will be really clear to you then.

Reply:
OK but then there are two issues:

1. We change the behavior of method 1 & method 2 which are not tested (we test method 3 only). Chances are small that somebody wants to keep Inf's in, but this could happen. E.g., if you use the old M0 method, there are Inf's at the border which you now replace by NaNs. Could even be good. But formally we need to test it, because method 1 & 2 are used in ExploreASL at other locations as method 3.
2. We clearly describe in the function header that not only NaNs but all non-real values are replaced.
3. We can avoid [1] by adding a warning if non-real non-nan voxels were found, and will be removed. Also fine with me.

Reply:
Henk - please check new code before answering. INF->NAN is applied beforehand, but only for METHOD 3 - 3 comments ago :)

1. Functionality of 1&2 is not affected. This is only applied selectively for METHOD 3
2. OK - I describe this for method in a header now.